src/account/BiconomySmartAccountV2.ts

```diff
@@ -1513,21 +1514,62 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
       userOp.maxFeePerGas = gasFeeValues?.maxFeePerGas as Hex
       userOp.maxPriorityFeePerGas = gasFeeValues?.maxPriorityFeePerGas as Hex
 
-      userOp = await this.getPaymasterUserOp(
-        userOp,
-        buildUseropDto.paymasterServiceData
-      )
-      return userOp
+      if(buildUseropDto.gasOffset){
+        userOp = await this.estimateUserOpGas(userOp)
+
+        const {verificationGasLimitIncrement, preVerificationGasIncrement, callGasLimitIncrement, maxFeePerGasIncrement, maxPriorityFeePerGasIncrement} = buildUseropDto.gasOffset;
+        userOp.verificationGasLimit = toHex(parseInt((Number(userOp.verificationGasLimit ?? 0) * convertToFactor(verificationGasLimitIncrement)).toString()));
+        userOp.preVerificationGas = toHex(parseInt((Number(userOp.preVerificationGas ?? 0) * convertToFactor(preVerificationGasIncrement)).toString()));
```

Comment:
what would be the cases / negative paths when these values will land up undefined here?

cause in all these cases it will just make resultant 0

Reply:
Are you talking about the offset values coming from args or about the userOp estimated values ? gasOffset values are handled in convertToFactor for undefined and userOp values are coming from bundler estimation so they should never be undefined or 0